app/Database/CourseQueries.hs

```diff
@@ -75,9 +75,18 @@ returnMeeting lowerStr sect session = do
                                  []
     return $ head entityMeetings
 
+-- | Convert Times into Time
+buildTime :: Times -> SqlPersistM Time
```

Comment:
1. Make this function have type signature `Times -> Time`, and move it to `Tables.hs`.
2. Make a function that converts the other way, e.g. `buildTimes :: Key Meeting -> Time -> Times`.

(In general, prefer pure functions; also, functions that convert between data types should be located where those data types are defined.)